### PR TITLE
Missing VkSparseMemoryBind VUIDs

### PR DESCRIPTION
**Pull request by aitor-lunarg**
While looking into sparse memory validation, found there were some VUIDs that would be helpful for what I'm currently looking into. They are related to VkSparseMemoryBind.

Added:
- VUID-VkSparseMemoryBind-size-01098
- VUID-VkSparseMemoryBind-resourceOffset-01099
- VUID-VkSparseMemoryBind-size-01100
- VUID-VkSparseMemoryBind-size-01102

Already implemented:
- VUID-VkSparseMemoryBind-memory-01097
- VUID-VkSparseMemoryBind-memoryOffset-01101 (moved to where we currently check all the similar ones)

Still missing:
- VUID-VkSparseMemoryBind-memory-01096 (planning to look into this one)
- VUID-VkSparseMemoryBind-memory-02730 (I can look into this one once finished with what I'm doing)
- VUID-VkSparseMemoryBind-memory-02731 (I can look into this one once finished with what I'm doing)

### CODE REVIEW

**Review by ziga-lunarg — APPROVED** [review]
LGTM, but it would be helpful if all the numbers were included in the logs.

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -15907,19 +15907,45 @@ bool CoreChecks::PreCallValidateGetEventStatus(VkDevice device, VkEvent event) c
...
         }
+
+        if (bind.memoryOffset >= mem_info->alloc_info.allocationSize) {
+            skip |= LogError(bind.memory, "VUID-VkSparseMemoryBind-memoryOffset-01101",
+                             "%s: %s memoryOffset is not less than the size of memory", func_name, parameter_name);
+        }
+
+        if ((mem_info->alloc_info.allocationSize - bind.memoryOffset) < bind.size) {
+            skip |= LogError(bind.memory, "VUID-VkSparseMemoryBind-size-01102",
+                             "%s: %s size must be less than or equal to the size of memory minus memoryOffset", func_name,
```
[the remark this answers is not in the document]
Comment (ziga-lunarg) [reply layers/core_validation.cpp:15928]: Other VUIDs as well

[the remark this answers is not in the document]
Comment (aitor-lunarg) [reply layers/core_validation.cpp:15928]: I'm unsure to which `numbers` you are refering to. If those numbers mean the indices of the structs, the `parameter_name` already has that information formatted before being fed to the function (lines 16014 and 16122 of this review)

[the remark this answers is not in the document]
Comment (ncesario-lunarg) [reply layers/core_validation.cpp:15928]: I think Ziga is referring to the values that led to this VUID getting triggered. So in this case, something like the following (in "printf style"):
```
"%s: %s size (" bind.size ") must be less than or equal to the size of memory (" mem_info->alloc_info.allocationSize ") minus memoryOffset (" bind.memoryOffset ")"
```

**Review by ncesario-lunarg — APPROVED** [review]
This LGTM pending resolution of Ziga's comments. If there isn't one already, can we get an issue that tracks the VUIDs you mention in the description of this PR?

File: layers/core_validation.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (aitor-lunarg) [reply layers/core_validation.cpp:15928]: That's definitely useful information for the user, I didn't think about it. Thank you for the suggestion!

**Comment by aitor-lunarg** [comment]
> This LGTM pending resolution of Ziga's comments. If there isn't one already, can we get an issue that tracks the VUIDs you mention in the description of this PR?

I'll create an issue, because I couldn't find any mentioning the ones I did in this PR.

**Review by Tony-LunarG — APPROVED** [review]
LGTM

**Pull request merged**